### ISSUE TITLE
Class model approach

### ISSUE THREAD

**Issue by yosbelms** [yosbelms/cor#2]
Javascript's OO model is very flexible, its makes every vendor define its own OO approach. Sometimes it is hard to make interaction between two libraries with different approaches. Libraries such as ExtJS, Backbone or Qooxdoo uses it's own feeling. So what approach do you think should Cor implement?

**Comment by yosbelms** [comment]
I've been working on this issue and my proposal right now is:

```
class Client {
    func init(name) {
        me.name = name
    }
}
```
- If there is _init_ method declaration it will compile to:

```
var Client = function Client
    (name) {var me = this;
        me.name = name;
    }
```

If there is no _init_, it should compile to the default initializer which accept either positional or key-value literal construction(see http://yosbelms.github.io/cor/docs/reference.html#literalconstructors):

```
class Client {   
        name = 'john'
}
```

it should compile to:

```
var Client = function Client(name) {
        // ...
        me.name = (name === undefined||name === null||$isConf)?'john':name;
    }
```

the resulting compilation is a little more complex than above, but the example is just enough to catch it.

**Comment by yosbelms** [comment]
Closed by commit: 964e508
